client/core/core.go

```diff
@@ -115,6 +121,25 @@ func (dc *dexConnection) markets() map[string]*Market {
 	return dc.marketMap
 }
 
+// getRegConfirms returns the number of confirmations received for the
+// dex registration or nil if the registration is completed
+func (dc *dexConnection) getRegConfirms() *uint32 {
+	dc.regConfMtx.RLock()
+	defer dc.regConfMtx.RUnlock()
+	if dc.regConfirms == regConfirmationsPaid {
+		return nil
+	}
+	return &dc.regConfirms
+}
+
+// setRegConfirms sets the number of confirmations received
+// for the dex registration
+func (dc *dexConnection) setRegConfirms(confs uint32) {
```

Comment:
Simple method to update `regConfirms` which deals with the synchronization internally.